src/App.tsx

```diff
@@ -1,204 +1,79 @@
-              ×
-            </button>
+  const [filterStatus, setFilterStatus] = useState<FilterStatus>(
+    FilterStatus.All,
+  );
 
-            {/* 'is-active' class puts this modal on top of the todo */}
-            <div data-cy="TodoLoader" className="modal overlay is-active">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
-            </div>
-          </div>
-        </section>
+  const filteredTodos = useMemo(
+    () =>
+      todos.filter(todo => {
+        if (filterStatus === FilterStatus.All) {
+          return true;
+        }
+
+        return filterStatus === FilterStatus.Completed
+          ? todo.completed
+          : !todo.completed;
+      }),
+    [todos, filterStatus],
+  );
 
-        {/* Hide the footer if there are no todos */}
-        <footer className="todoapp__footer" data-cy="Footer">
-          <span className="todo-count" data-cy="TodosCounter">
-            3 items left
-          </span>
+  const todosLeftNum = useMemo(
+    () => todos.filter(todo => !todo.completed).length,
+    [todos],
+  );
 
-          {/* Active link should have the 'selected' class */}
-          <nav className="filter" data-cy="Filter">
-            <a
-              href="#/"
-              className="filter__link selected"
-              data-cy="FilterLinkAll"
-            >
-              All
-            </a>
+  useEffect(() => {
+    (async () => {
+      try {
+        const data = await getTodos();
 
-            <a
-              href="#/active"
-              className="filter__link"
-              data-cy="FilterLinkActive"
-            >
-              Active
-            </a>
+        setTodos(data);
+      } catch (err) {
+        setErrorMessage(ErrorType.LoadTodos);
+      }
+    })();
+  }, []);
 
-            <a
-              href="#/completed"
-              className="filter__link"
-              data-cy="FilterLinkCompleted"
-            >
-              Completed
-            </a>
-          </nav>
+  return (
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
 
-          {/* this button should be disabled if there are no completed todos */}
-          <button
-            type="button"
-            className="todoapp__clear-completed"
-            data-cy="ClearCompletedButton"
-          >
-            Clear completed
-          </button>
-        </footer>
+      <div className="todoapp__content">
+        <TodoHeader />
+
+        {todos.length > 0 && (
+          <>
+            <section className="todoapp__main" data-cy="TodoList">
+              {filteredTodos.map(todo => (
+                <TodoItem key={todo.id} todo={todo} />
+              ))}
+            </section>
+            <TodoFooter
+              filterStatus={filterStatus}
+              setFilterStatus={setFilterStatus}
+              todosLeft={todosLeftNum}
+            />
+          </>
+        )}
```

Comment:
According to the task checklist, you should hide the list and the footer if there are no todos yet. Currently, the code only checks if `todos.length > 0` to render the list and footer, but it doesn't explicitly hide them when there are no todos. Consider adding logic to ensure these elements are hidden when the todos array is empty .